### PR TITLE
chore: enroll validators for megabatch + infra deployer fixes

### PR DESCRIPTION
**Pull request by paulbalaji**
chore: enroll validators for megabatch

- ~~also includes some of the verifier fixes in https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/4381~~
- drive-by: passing the timeout through to the core deployer
- drive-by: fixing artifact write-back in infra deploy (resolves https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/4351)
- drive-by: adds a script to solely check for validator announcements and print a nice table
image

### CODE REVIEW

File: typescript/infra/scripts/check-validator-announce.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import { defaultMultisigConfigs } from '@hyperlane-xyz/sdk';
+import { eqAddress } from '@hyperlane-xyz/utils';
+
+import { isEthereumProtocolChain } from '../src/utils/utils.js';
+
+import { getArgs, withChains } from './agent-utils.js';
+import { getEnvironmentConfig, getHyperlaneCore } from './core-utils.js';
+
+async function main() {
```
Comment (Mo-Hussain) [inline typescript/infra/scripts/check-validator-announce.ts:9]: Could be nice to extend this to check is the validators have setup their S3 bukcet correctly and are signing the latest check point https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/cab86f2f971c2c03f7620101134f6fb2f06690c5/typescript/cli/src/validator/preFlightCheck.ts#L51

File: typescript/sdk/src/core/HyperlaneCoreDeployer.ts [hunk 2]
```diff
@@ -34,10 +34,11 @@ export class HyperlaneCoreDeployer extends HyperlaneDeployer<
     readonly ismFactory: HyperlaneIsmFactory,
     contractVerifier?: ContractVerifier,
     concurrentDeploy: boolean = false,
+    chainTimeoutMs: number = 1000 * 60 * 10, // 10 minutes
```
Comment (Mo-Hussain) [inline typescript/sdk/src/core/HyperlaneCoreDeployer.ts:37]: Nice!

File: typescript/infra/scripts/check-validator-announce.ts (hunk 1 above)
Comment (Mo-Hussain) [reply typescript/infra/scripts/check-validator-announce.ts:9]: I should probably refactor the code so it can be reused

Comment (paulbalaji) [reply typescript/infra/scripts/check-validator-announce.ts:9]: there's already a script for that - but I wanted a very specific announce-only check for now

Comment (paulbalaji) [reply typescript/infra/scripts/check-validator-announce.ts:9]: what I'd like really is the CLI command for checking validators, but to be able to operate that over the set of default multisigs we track

File: typescript/infra/scripts/print-balances.ts [hunk 3]
```diff
@@ -38,10 +43,25 @@ async function main() {
...
           await Promise.all(Object.values(keys).map((key) => key.fetch()));
-          if (keys[chain]) {
-            const balance = await provider.getBalance(keys[chain].address);
+
+          // Default to known deployer/relayer addresses if not found
+          let address = keys[chain]?.address;
+          if (!address) {
+            if (role === Role.Deployer) {
+              address =
+                environment === 'mainnet3' ? MainnetDeployer : TestnetDeployer;
```
[the remark this answers is not in the document]
Comment (paulbalaji) [reply typescript/infra/scripts/print-balances.ts:55]: which enum? the only env enum I can see is `AgentEnvironment` which is only `mainnet` or `testnet`, but the infra scripts care more about which of our envs `mainnet3`/`testnet4`/`test`

File: typescript/infra/src/deployment/deploy.ts
Comment (Mo-Hussain) [inline typescript/infra/src/deployment/deploy.ts:1]: Should we remove `process.on('SIGINT', async () => {` on ln39?

File: typescript/infra/scripts/print-balances.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (Mo-Hussain) [reply typescript/infra/scripts/print-balances.ts:55]: I thought there was an enum for `mainnet3/testnet4/test` my bad

File: typescript/infra/src/deployment/deploy.ts
Comment (paulbalaji) [reply typescript/infra/src/deployment/deploy.ts:1]: ahh thanks, probably got added back in a refactor

File: typescript/infra/scripts/print-balances.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (paulbalaji) [reply typescript/infra/scripts/print-balances.ts:55]: would be good to have it for sure - can add it to my backburner refactor tasks

File: typescript/infra/scripts/check-validator-announce.ts (hunk 1 above)
Comment (paulbalaji) [reply typescript/infra/scripts/check-validator-announce.ts:9]: see existing script: 
- https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/330d302a6181191e61141786c0999d7200ab6da4/typescript/infra/scripts/verify-validators.ts#L33
- https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/330d302a6181191e61141786c0999d7200ab6da4/typescript/infra/src/agents/aws/validator.ts#L48

which iterates through all storage locations onchain and also checks their checkpoint/mailbox/etc

**Review by Mo-Hussain — APPROVED** [review]

**Pull request merged**